src/components/JoinList.js

```diff
@@ -1,36 +1,45 @@
-import React, { useEffect } from 'react';
+import React from 'react';
 import { Link } from 'react-router-dom';
 import '../pages/HomePage.css';
 import useListToken from '../useListToken';
 
 const JoinList = () => {
-  const { token, saveToken } = useListToken();
-  useEffect(() => {
-    saveToken(token);
-  }, [saveToken, token]);
+  const { saveToken } = useListToken();
+
+  const [joinToken, setJoinToken] = React.useState('');
```

Comment:
If you implement the comment above about the import statement, you could then change `React.useState('');` to just `useState('');`.

A better initial state here might be `null` which is commonly used to denote something that is purposely set to an empty value.